### DISCUSSION TITLE
@ifBuildProperty for qute templates?

### DISCUSSION THREAD

**Discussion by TonyWeston** [quarkusio/quarkus#32939] [Q&A] [answered]
I want a particular endpoint/qute template to be available only when built with a particular build property.  My qute template references a CDI bean,  however in the bean I have defined it as

```
@Named("warningMatrix")
@IfBuildProperty(stringValue = "true", name="matrixEnabled")
public class MVWarningMatrix {
```

This removes the CDI bean, as expected, if build property matrixEnabled is not set to true.   However, the build fails because the qute template that references this bean, is unable to find it at build time.

Is there a way of specifying @IfBuildProperty against the qute template also, since this is not required either.

[a comment by an automated account is not shown]

**Reply by mkouba** [reply]
Unfortunately, there's no way to tell Qute to skip the validation of the `cdi:warningMatrix` value expressions.

> since this is not required either

Does it mean that the template is not used at all if `matrixEnabled=false`?

**Reply by tonywestonuk** [reply]
>Does it mean that the template is not used at all if matrixEnabled=false?

Yes. Ideally there would be somewhere to add configuration, to prevent the template being included in the deployment, if matrixEnabled=false.

**Reply by mkouba** [reply, marked as answer]
We have the [`quarkus.qute.template-path-exclude`](https://quarkus.io/guides/qute-reference#quarkus-qute_quarkus.qute.template-path-exclude) config property but you would have to pass the config value along with `matrixEnabled=false`. For example `-DmatrixEnabled=false -Dquarkus.qute.template-path-exclude=myTemplate`.

**Reply by TonyWeston** [reply]
That'll do!  Thanks.

**Reply by mkouba** [reply]
Ok. Keep in mind that it's a regular expression...
